Options/Applicative/Types.hs

```diff
@@ -188,16 +188,16 @@ instance Functor CReader where
 
 -- | An 'OptReader' defines whether an option matches an command line argument.
 data OptReader a
-  = OptReader [OptName] (CReader a) ParseError          -- ^ option reader
-  | FlagReader [OptName] !a                             -- ^ flag reader
-  | ArgReader (CReader a)                               -- ^ argument reader
+  = OptReader (Maybe String) [OptName] (CReader a) ParseError -- ^ option reader
+  | FlagReader (Maybe String) [OptName] !a                    -- ^ flag reader
+  | ArgReader (Maybe String) (CReader a)                      -- ^ argument reader
   | CmdReader (Maybe String)
               [String] (String -> Maybe (ParserInfo a)) -- ^ command reader
```

Comment:
Having a `(Maybe String)` on every constructor seems a bit rough. It should probably be moved to the `Option` type.